Src/AutoFixture/Kernel/FactoryMethodQuery.cs

```diff
@@ -39,6 +43,7 @@ public IEnumerable<IMethod> SelectMethods(Type type)
             return from mi in type.GetMethods(BindingFlags.Static | BindingFlags.Public)
                    where mi.ReturnType == type
                    let parameters = mi.GetParameters()
+                   where mi.GetParameters().All(p => p.ParameterType != type)

```

Comment:
It's a good point, I think.

I'm not sure if we should consider it to be a _Factory_ method when it requires an instance of itself as the only argument, it seems not.